src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/ClaimsController.cs

```diff
@@ -85,4 +88,143 @@ public async Task<ActionResult<ResultList<Claim>>> GetClaims([FromServices] IUni
                 })
         };
     }
+
+    /// <summary>
+    /// Queues a request to claim two certificate for a given quantity.
+    /// </summary>
+    /// <param name="bus">The masstransit bus to queue the request to</param>
+    /// <param name="request">The claim request</param>
+    /// <response code="202">Claim request has been queued for processing.</response>
+    /// <response code="401">If the user is not authenticated.</response>
+    [HttpPost]
+    [Route("v1/claims")]
+    [Produces("application/json")]
+    [ProducesResponseType(typeof(ClaimResponse), StatusCodes.Status202Accepted)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    public async Task<ActionResult<ClaimResponse>> ClaimCertificate(
+        [FromServices] IBus bus,
+        [FromBody] ClaimRequest request
+    )
+    {
+        if (!User.TryGetSubject(out var subject)) return Unauthorized();
+
+        var command = new ClaimCertificateCommand
+        {
+            Owner = subject,
+            ClaimId = Guid.NewGuid(),
+            ConsumptionRegistry = request.ConsumptionCertificateId.Registry,
+            ConsumptionCertificateId = request.ConsumptionCertificateId.StreamId,
+            ProductionRegistry = request.ProductionCertificateId.Registry,
+            ProductionCertificateId = request.ProductionCertificateId.StreamId,
+            Quantity = request.Quantity,
+        };
+
+        await bus.Publish(command);
```

Comment:
Should this be awaited?

Reply:
Yes, to make sure it happens on the same thread and before the response is returns to the caller.